File: 1-introdução/Livro/src/class/Livro.ts


export class Livro {
    titulo: string
    autor: string
    anoPublicacao: number

    constructor(titulo: string, autor: string, anoPublicacao: number) {
        this.titulo = titulo
        this.autor = autor
        this.anoPublicacao = anoPublicacao
    }

    descricao(){
        return `Livro: ${this.titulo}, por ${this.autor} ano ${this.anoPublicacao}`
    }
}

File: 3-POO-LOGIC-WITH-PYTHON/1-INTRODUCAO/CLASSE-LIVRO/classes/Livro.py

class Livro:
    def __init__(self, nome: str, autor: str, ano_publicacao: int) -> None:
        self.nome = nome
        self.autor = autor
        self.ano_publicacao = ano_publicacao
    
    def descricao(self) -> str:
       return f"Livro: {self.nome} por {self.autor} ano de {self.ano_publicacao}" 